pkg/apis/serving/v1alpha1/configuration_conversion_test.go

```diff
@@ -0,0 +1,218 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	duckv1beta1 "github.com/knative/pkg/apis/duck/v1beta1"
+	"github.com/knative/pkg/ptr"
+	"github.com/knative/serving/pkg/apis/serving/v1beta1"
+)
+
+func TestConfigurationConversionBadType(t *testing.T) {
+	good, bad := &Configuration{}, &Service{}
+
+	if err := good.ConvertUp(context.Background(), bad); err == nil {
+		t.Errorf("ConvertUp() = %#v, wanted error", bad)
+	}
+
+	if err := good.ConvertDown(context.Background(), bad); err == nil {
+		t.Errorf("ConvertDown() = %#v, wanted error", good)
+	}
+}
+
+func TestConfigurationConversion(t *testing.T) {
+	tests := []struct {
+		name     string
+		in       *Configuration
+		badField string
+	}{{
+		name: "simple configuration",
+		in: &Configuration{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:       "asdf",
+				Namespace:  "blah",
+				Generation: 1,
+			},
+			Spec: ConfigurationSpec{
+				Template: &RevisionTemplateSpec{
+					Spec: RevisionSpec{
+						RevisionSpec: v1beta1.RevisionSpec{
+							PodSpec: v1beta1.PodSpec{
+								ServiceAccountName: "robocop",
+								Containers: []corev1.Container{{
+									Image: "busybox",
+									VolumeMounts: []corev1.VolumeMount{{
+										MountPath: "/mount/path",
+										Name:      "the-name",
+										ReadOnly:  true,
+									}},
+								}},
+								Volumes: []corev1.Volume{{
+									Name: "the-name",
+									VolumeSource: corev1.VolumeSource{
+										Secret: &corev1.SecretVolumeSource{
+											SecretName: "foo",
+										},
+									},
+								}},
+							},
+							TimeoutSeconds:       ptr.Int64(18),
+							ContainerConcurrency: 53,
+						},
+					},
+				},
+			},
+			Status: ConfigurationStatus{
+				Status: duckv1beta1.Status{
+					ObservedGeneration: 1,
+					Conditions: duckv1beta1.Conditions{{
+						Type:   "Ready",
+						Status: "True",
+					}},
+				},
+				ConfigurationStatusFields: ConfigurationStatusFields{
+					LatestReadyRevisionName:   "foo-00002",
+					LatestCreatedRevisionName: "foo-00009",
+				},
+			},
+		},
+	}, {
+		name:     "cannot convert build",
+		badField: "build",
+		in: &Configuration{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:       "asdf",
+				Namespace:  "blah",
+				Generation: 1,
+			},
+			Spec: ConfigurationSpec{
+				DeprecatedBuild: &RawExtension{
+					Object: &Revision{},
+				},
+				Template: &RevisionTemplateSpec{
```

Comment:
Do you need to provide whole template, to just get the error from the build being unconvertable?

Reply:
I simplified it, we don't need one this complex.